README.md

```diff
@@ -145,7 +174,8 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+GrammarParser.py
+run with: py GrammarParser.py to have tests run
```

Comment:
# 5

You've written a nice parser for the grammar, but it isn't a state machine: it's just linear code. Clearly the design is informed by the state transitions, but there's no explicit FSM.

I was looking for something like:

~~~ python
import unittest
from functools import reduce
ERROR  = "ERROR"
OK     = "OK"

states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 5, "noisily": 5, "EOI": OK },
  5:  { "EOI":    OK }
}

def getNextState(currentState, nextWord):
    return states.get(currentState,{}).get(nextWord, ERROR)

def stateMachine(sentence):
    return reduce(getNextState, sentence, 0) == OK
~~~